Docs proofreading (#436)
slskd/slskd: docs/config.md

## Environment Variables

Environment variables are loaded after defaults. Each environment variable name is prefixed with `SLSKD_` (this prefix is omitted for the remainder of this documentation).

Environment variables are ideal for use cases involving Docker where configuration is expected to be changed neither often nor remotely.

Some options are backed by arrays or lists and allow multiple options to be set via a single environment variable. To achieve this, separate each list item by a semicolon `;` in the value:

```
SLSKD_SOME_OPTION=1;2;3
```

## YAML Configuration File

Configuration is loaded from the YAML file located at `<application directory>/slskd.yml` after environment variables. The location can be changed by specifying the path in the `CONFIG` environment variable or `--config` command-line argument.

The application watches for changes in the YAML file and will reload the configuration when they are detected. Options will be updated in real-time and transmitted to the web UI. If a server reconnect or application restart is required for changes to take effect fully, a flag will be set indicating so.

> The YAML file can be read and written at run time via API calls or edited on a disk.

Reviewer: ```suggestion
The YAML file can be read and written at run time via API calls or edited on disk.
```

> If no such configuration file exists at startup, the example file `/config/slskd.example.yml` is copied to this location for convenience.

Reviewer: you can tell i didn't use a spell checker :)

The application directory configuration option determines the location of the YAML file, the default locations of the download and incomplete directories, and the location of application working data, such as logs and SQLite databases.

Because the location of the YAML file derived from this value, it can't be specified within the YAML file and must instead be set with the `APP_DIR` environment variable or `--app-dir` command-line argument.

If no value is specified, the location defaults to either `~/.local/share/slskd` (on Linux and macOS) or `%localappdata%/slskd` (on Windows).

This value is set to `/app` within the official Docker image.

# Other Directory Configuration

## Incomplete and Downloads

> By default, the incomplete and downloaded files are saved in `APP_DIR/incomplete` and `APP_DIR/downloads` directories, respectively. The application will create these directories on startup if they don't exist.

Reviewer: ```suggestion
By default, incomplete and downloaded files are saved in `APP_DIR/incomplete` and `APP_DIR/downloads` directories, respectively. The application will create these directories on startup if they don't exist.
```

Any number of shared directories can be configured.

Paths must be absolute, meaning they must begin with `/`, `X:\`, or `\\`, depending on the system. Relative paths, such as `~/directory` or `../directory`, and root paths (e.g., an entire drive, mount, or network share) are not supported.

> Shares can be excluded by prefixing them with `-` or `!`. It is useful in situations where sharing a subdirectory of a share isn't desired, for example, if a user wants to share their entire music library but not their personal recordings:

Reviewer: ```suggestion
Shares can be excluded by prefixing them with `-` or `!`. This is useful in situations where sharing a subdirectory of a share isn't desired, for example, if a user wants to share their entire music library but not their personal recordings:
```

## Distributed Network

> Options for the Soulseek distributed network are how search requests are delivered.

Reviewer: ```suggestion
Options for the Soulseek distributed network, which is how search requests are delivered.
```

The port on which the application listens for incoming connections.

> As with any other Soulseek client, configuring the listen port and port forwarding ensures full connectivity with other clients, including those not correctly configured a listening port.  

Reviewer: ```suggestion
As with any other Soulseek client, configuring the listen port and port forwarding ensures full connectivity with other clients, including those without a correctly configured a listening port.  
```

### Timeouts

> Timeout options control how long the application waits for connections and how long connections can be inactive before they are disconnected.

Reviewer: ```suggestion
Timeout options control how long the application waits for connections to connect, and how long connections can be inactive before they are disconnected.
```

This is a nuanced one, since a wait for the connection to connect and a wait for the connection to close/do work are different.

> Higher connect timeout values will help ensure that operations (browse, download requests, etc.) are successful the first time but decrease the responsiveness of commands that ultimately fail.

Reviewer: ```suggestion
Higher connect timeout values will help ensure that operations (browse, download requests, etc.) are successful the first time but decrease the responsiveness of commands that will ultimately fail.
```

'will' is like 'are destined to'

### Buffers

> Buffer options control the application's internal buffer to batch socket reads and writes. The OS controls actual socket buffer sizes (.NET is not great in this department currently, and the behavior is not consistent cross-platform).

Reviewer: ```suggestion
Buffer options control the application's internal buffer to batch socket reads and writes. The host Operating System controls actual socket buffer sizes (.NET is not great in this department currently, and the behavior is not consistent cross-platform).
```

probably should have been spelled out from the start
